Chatbot.py

```diff
@@ -21,17 +24,19 @@
     )
     b.form_submit_button("Send", use_container_width=True)
 
-for msg in st.session_state.messages:
-    message(msg["content"], is_user=msg["role"] == "user")
+for idx, msg in enumerate(st.session_state.messages):
+    message(msg["content"], is_user=msg["role"] == "user", key=idx)
```

Comment:
What does this change do, or why change it?

Reply:
streamlit chat was raising a duplicate widget error without this, maybe because I happened to use the same message twice in a row

Reply:
ah OK - guessing it won't be a problem after we move off the custom component to native chat UI, I will probably revert this at that point unless you have any concerns. cc @sfc-gh-lmasuch